### DISCUSSION TITLE
Cobbler is incomplete without cobbler-web

### DISCUSSION THREAD

**Discussion by asimat** [cobbler/cobbler#3125] [General]
I have introduced cobbler to 8 or more companies in the last 8 years or so. Currently I am still using cobbler 3.2 with cobbler-web, which works nicely with puppet-7 as I did do some manual changes. Recently also got Rocky and Alma to work with it.

My customers are small to medium companies who do not have many linux engineers and mostly Juniors use cobbler to deploy handle physical computers/desktop. As its mostly Juniors using it so web interface is an excellent tool for them.

I have tested the Angular web interface but its still not Production-Ready, actually it launches but could not get it working with cobbler and after looking at the GUI I didn't want to use it anyways.

The companies where I work, want to keep software up to date and Auditors don't like us using outdated software. e.g. cobbler-web 2.8 fails the vulnerability test, so I guess by end of this year cobbler 3.2 might also not be allowed for us to use.

In short, we will not be using cobbler if cobbler-web is not working. So would like to know if someone is working on cobbler-web side of things or not ?

Also the guide for cobbler-web only shows how to run test server, accessible from localhost only. Angular being too new for me, I won't have a clue how to change anything. So a better guide to get it working for small production setup is needed urgently.

Can we please spend some resources on cobbler-web and make it usable for people like me to get involved and contribute.

**Comment by SchoolGuy** [comment]
What resources do you need? I am willing to help to get you started but at the moment my company does not allow me to put priority on this. I understand that you want a webinterface and I would love to support that but the bandwith as a single developer is not there atm. The reason I cut out the Django Webinterface was that it was preventing changes that were required to fix outstanding issues with the core code.

**Reply by trevormac789** [reply]
Hi, this sort of answers my question. So you completely removed the webui until version 4? Could you make that a little clearer in the documentation that it's been removed. I've been troubleshooting for 30 minutes and gathered that it's just not there, but it wasn't completely clear. I thought maybe you were saying that django is there until version 4 replaces it.  

So the only option is Angular at the moment? I'm willing to give that a try. Will that be what's eventually merged into production?

Thank you.

**Reply by SchoolGuy** [reply]
@trevormac789 Well initially the plan was to have a drop-in replacement ready by 3.3 but this didn't happen due to the conflict of interests at work and upstream interests (simply spoken: there was no time).

I believe it is very clear that I have removed it due to this comment: <https://cobbler.readthedocs.io/en/latest/user-guide.html#web-interface>. If you believe there is a better or more prominent place then please let me know. Feel free to also open a PR and we discuss your proposed changes there!

To answer your last question: Yes I have chosen Angular because of data I collected and personal experience. The repo can be found here: <https://github.com/cobbler/cobbler-web/>. If you need help, I am more then welcome to give you a hand for getting started. That repository will also be the final place for the new web ui since I want to keep the web ui completely separate from the server.

**Reply by trevormac789** [reply]
Okay. Thanks SchoolGuy. Sorry, I probably misread it and overlooked that you stated that the Web UI had been removed. I sometimes skim and overlook the details. After rereading the OP, I rolled back to 3.2 as it's easier for me and I eagerly will await version 4. I can now login.

Thank you very much for your hard work on the project. We've been using cobbler in conjunction with puppet for years and it's a life saver for our team. 

I know you are very stretched, but do you at least have a rough ETA for version 4? Our server is internal so we're not as concerned about security on the server and 3.2 will suffice for now, but I actually think a facelift for the Web UI would be a welcome change. 

Thanks again!

**Reply by SchoolGuy** [reply]
So an ETA is not doable sadly. My plans always failed in the last two years for various reasons in my private life. Version 4.0 will take at least two years I would say since things like an HTTP API and full containerizations are easy to do but in a single-person manner on private time, I can't invest more then 5h a week at the moment on average. The time I spent at work on Cobbler are mostly related to fulfilling SUSE Manager customer related things that indirectly benefit the project but don't help finish the WebUI. I will be able to tell you more once I have taken my yearly vacation. The plan at the moment is to sacrifice those two weeks into the WebUI and bug squashing on the server side.
